sequence_processing_pipeline/Pipeline.py

```diff
@@ -83,38 +160,31 @@ def __init__(self, configuration_file_path, run_id, sample_sheet_path,
 
         self.pipeline_type = pipeline_type
 
-        if config_dict:
-            if 'configuration' in config_dict:
-                self.configuration = config_dict['configuration']
-                self.configuration_file_path = None
-            else:
-                raise PipelineError(f"{config_dict} does not contain the "
-                                    "key 'configuration'")
-        else:
-            self.configuration_file_path = configuration_file_path
-            try:
-                f = open(configuration_file_path)
-                self.configuration = json_load(f)['configuration']
-                f.close()
-            except TypeError:
-                raise PipelineError('configuration_file_path cannot be None')
-            except FileNotFoundError:
-                raise PipelineError(f'{configuration_file_path} does not '
-                                    'exist.')
-            except JSONDecodeError:
-                raise PipelineError(f'{configuration_file_path} is not a '
-                                    'valid json file')
+        self.configuration_file_path = configuration_file_path
+
+        # along with configuration profiles, a 'general' configuration file
+        # is needed to provide the search paths to the run-directories. These
+        # are used to locate the run-directory specified and parse the required
+        # files to select the right configuration profile.
+        try:
+            f = open(configuration_file_path)
+            self.configuration = json_load(f)
+            f.close()
+        except TypeError:
+            raise PipelineError('configuration_file_path cannot be None')
+        except FileNotFoundError:
+            raise PipelineError(f'{configuration_file_path} does not '
+                                'exist.')
+        except JSONDecodeError:
+            raise PipelineError(f'{configuration_file_path} is not a '
+                                'valid json file')
```

Comment:
Are there tests for these cases?

Reply:
See: 

https://github.com/charles-cowart/mg-scripts/blob/1bb127c1b9dd2c1ca65398942421c10a7eea2e01/sequence_processing_pipeline/tests/test_Pipeline.py#L202

https://github.com/charles-cowart/mg-scripts/blob/1bb127c1b9dd2c1ca65398942421c10a7eea2e01/sequence_processing_pipeline/tests/test_Pipeline.py#L191

I'm not sure if a test for json.load() is needed since it's such a thin wrap but I'll add one at the bottom of the test method.